### DISCUSSION TITLE
For audio containing native English speakers is "tiny" mode more than sufficient?

### DISCUSSION THREAD

**Discussion by roschler** [openai/whisper#784] [Q&A] [answered]
My PC unfortunately has an OpenCL card, not CUDA, so I have to use the `tiny` model since the `medium` model is glacially slow when launching Whisper in `cpu` mode.  However, the results of the `tiny` model when run on an audio file with native English speakers is very good, so I'm wondering if the `medium` mode would really be that much better to the point that I would want to build a new PC with a CUDA card?

**Comment by Hunter-Dolan** [comment, marked as answer]
In my experience, the larger models are significantly better at dealing with edge cases. If your audio isn't running into edge cases (poor audio quality, connecting cutting out, stuttering, 'umms' and 'ahhs', overlapping speech, etc), and you expect only native English speakers to be using it, you're probably fine with the tiny model.

You can evaluate these models pretty easily on huggingface. Here is the largest one https://huggingface.co/openai/whisper-large-v2, a demo is on the right side of the page. And here is medium https://huggingface.co/openai/whisper-medium

If you do find the medium or large models useful, personally I wouldn't build a new PC anymore. The hardware for a ML Development Machine is coming out faster than it makes economical sense to purchase, and you can get access to better (datacenter grade) GPUs for a dollar an hour with LambdaLabs, CoreWeave, etc. I just use VS Code Remote to SSH in and it's as if my laptop has an A100 running on it.

Hope this helps!

**Reply by roschler** [reply]
Thanks!  I'm finding that even with the `tiny` model even modest sized audio files take a long time to process.  Would those third party GPU systems be much faster?  By "much faster" I mean in the 2X to 3X or greater speed improvement range.

**Reply by Hunter-Dolan** [reply]
@roschler In my experience, the datacenter GPUs (NVIDIA A100, A40, etc) have significantly better throughput versus the consumer grade GPUs. A 3x to 4x improvement over a lower end GPU isn't unreasonable so long as you're comparing the same model size (tiny on a consumer grade GPU still will probably be faster than large on a datacenter grade GPU). In addition you have significantly more GPU memory which allows you to run extremely large models. And if you're comparing CPU performance to GPU performance, it'll be night and day.

**Reply by roschler** [reply]
Thanks again.  One final question, when I run the Whisper client, is all the processing happening on my PC, or is the client talking to some process that exists only on OpenAI's servers and if the latter is true, what is that processing doing? And, do you have a favorite GPU cluster service with reasonable rates?  I've heard of people saving a lot of money with AWS "spot instances", but if a GPU service made the task allocation across clusters effortless or automatic, I'd spring for that service over spot instances.  I'm assuming that with AWS spot instances I'd have to configure and do the cross PC task allocation myself (not sure, just guessing).
